Assets/Scripts/PlayerController.cs

```diff
@@ -12,16 +12,73 @@ public class PlayerController : MonoBehaviour
 
     private Vector3 _forward;
     private Vector3 _right;
-
-    private bool _onGround = true;
+    private Vector3 _heading;
 
     private Rigidbody _rb;
+    public BoxCollider Bc;
+    private Vector3[] groundSkinVertices = new Vector3[16];
+    private Vector3[] forwardSkinVertices = new Vector3[16];
 
     // Use this for initialization
     private void Start()
     {
         _rb = GetComponent<Rigidbody>();
+        _heading = Vector3.forward;
+        groundSkinVertices[0] = Vector3.zero - (Vector3.right * Bc.bounds.extents.x) - (Vector3.forward * Bc.bounds.extents.z);
+        groundSkinVertices[1] = Vector3.zero - (Vector3.right * Bc.bounds.extents.x) + (Vector3.forward * Bc.bounds.extents.z);
+        groundSkinVertices[2] = Vector3.zero + (Vector3.right * Bc.bounds.extents.x) + (Vector3.forward * Bc.bounds.extents.z);
+        groundSkinVertices[3] = Vector3.zero + (Vector3.right * Bc.bounds.extents.x) - (Vector3.forward * Bc.bounds.extents.z);
```

Comment:
this can be changed to 
`Vector3(Bc.bounds.extents.x, -Bc.bounds.extents.z)`

Same for the rest.